src/client.rs

```diff
@@ -1010,14 +1010,14 @@ where
                 // SET SHARD TO
                 Some((Command::SetShard, _)) => {
                     // Selected shard is not configured.
-                    if query_router.shard() >= pool.shards() {
+                    if query_router.shard().unwrap_or(0) >= pool.shards() {
                         // Set the shard back to what it was.
                         query_router.set_shard(current_shard);
 
                         error_response(
                             &mut self.write,
                             &format!(
-                                "shard {} is more than configured {}, staying on shard {} (shard numbers start at 0)",
+                                "shard {:?} is more than configured {}, staying on shard {:?} (shard numbers start at 0)",
```

Comment:
We shouldn't leak debug rendering into client responses. I'd prefer we unwrap here if that's the path we're taking with `Option`.